app/javascript/components/ArtistTransactions.jsx

```diff
@@ -0,0 +1,96 @@
+import PropTypes from "prop-types";
+import React from "react";
+import CreateTransaction from "./CreateTransaction";
+import ReactModal from "react-modal";
+import Currency from 'react-currency-formatter';
+
+/**
+* @prop artist: artist associated with transactions
+*/
+
+class ArtistTransactions extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      show: false,
+      transactions: [],
+      didMount: false
+    }
+  }
+
+  componentDidMount = () => {
+    const artist_id = this.props.artist.id;
+    const transactions_route = APIRoutes.artists.transactions(artist_id);
+    Requester.get(transactions_route).then(
+      response => {
+        this.setState({
+          transactions: response,
+          didMount: true
+         });
+      },
+      error => {
+        console.error(error);
+      }
+    );
+  }
+
+  showModal = () => {
+    this.setState({ show: true });
+  };
+
+  hideModal = () => {
+    this.setState({ show: false });
+  };
+
+  render() {
+    if (!this.state.didMount) {
+      return (
+        <div />
+      );
+    }
+    return (
+      <div className="transactions-page">
+        <h2 className="f4 lh-title fw6">Transactions</h2>
+        <br />
+        {this.state.transactions.map(transaction => (
+          <div className="bb b--gray" key={transaction.id}>
+            <p>
+              <span className="b">Title</span>
+              <span className="i"> {transaction.work_title} </span>
+              <span>
+                <Currency
+                  quantity={transaction.price}
+                  symbol="$"
+                  locale="en"
+                />
+              </span>
+            </p>
+            <p>
+              <span className="b">Patron</span>
+              <span className="i"> {transaction.buyer_name} </span>
+              <span className="i">{transaction.transaction_type}</span>
+            </p>
+            <p>
+              <span>{transaction.comment}</span>
+            </p>
+          </div>
+        ))}
+
+        <ReactModal className="ba pa3 modal" isOpen={this.state.show} onRequestClose={this.hideModal}>
```

Comment:
I like the way you structured this. However, I have a couple of suggestions:

- All modals contain shared options such as the submit and cancel buttons, as well as shared styling. We should create a separate `StyledModal` class that simply accepts the contents through child components to create another level of abstraction, with submit/cancel buttons & props. From there, additional modal components can extend the `StyledModal` class
- It seems extraneous to store in the state whether the modal is closed or open. Truly, the parent component doesn't need to know whether the modal is closed or open. Instead, we should store this state within the `StyledModal` class.
- A modal component holistically is a button that upon clicked displays the actual modal. Thus, you can render the button within the `StyledModal` class and use that to manage displaying the modal. Then, you can import the component, and it will simply display as a button normally

Reply:
Also, this ensures modularity! Then we can extend the class for other uses

Reply:
A good example to look at would be SJC's